### ISSUE TITLE
Error accessing `request.promster.start` in hapi

### ISSUE THREAD

**Issue by 2color** [tdeekens/promster#150]
**Describe the bug**
- Error accessing the `start` in the `request` callback: https://github.com/tdeekens/promster/blob/master/packages/hapi/modules/plugin/plugin.js#L49

**To Reproduce**

1. Given a server as follows:
```js
  import { createPlugin } from '@promster/hapi'
  import { createServer as createMetricsServer } from '@promster/server'
  import Hapi from '@hapi/hapi'

  const server = Hapi.server({
    port: 4000,
    host: process.env.HOST,
  })

  await server.register(createPlugin())

  server.route({
    method: 'GET',
    path: '/',
    handler: (request, h) => {
      return 'hello!'
    }
  })

  await server.start()
  console.log('Server running on %s', server.info.uri)

  await createMetricsServer({ port: 4001 })
  console.log(`@promster/server started on port ${process.env.METRICS_PORT}.`)

  await server.register(createPlugin())
```
2. Then 'curl localhost:4000'
```
TypeError: Cannot read property 'start' of undefined
    at Object.onResponseHandler [as listener] (/...../myserver/node_modules/@promster/hapi/modules/plugin/plugin.js:54:40)
```

**Expected behavior**
- No error and `request.promster` available in `onResponseHandler`

**Additional context**
- Running:
  - hapi.js `18.3.1`
  - `@promster/hapi 2.2.4`
  - `@promster/server 2.1.1`

- [According to the docs](https://hapijs.com/api#-request-event) the `request`-event callback uses the `function(request, event, tags)` signature rather than `(request, h) => {}` as in https://github.com/tdeekens/promster/blob/47a4f1d4352a040e1516eca238f524d25c2f633e/packages/hapi/modules/plugin/plugin.js#L48. 
- [According to the docs](https://hapijs.com/api#-requestplugins) the `request.plugins` is the place to set things like the `promster` object:
> Plugin-specific state. Provides a place to store and pass request-level plugin data. The
- Strangely, when debugging, the `onRequestHandler` wouldn't get called on incoming requests.

**Comment by tdeekens** [comment]
Thanks for the very nice but report and sorry for the inconvienence. I will try to look into this as soon as I find time.

**Comment by tdeekens** [comment]
Turns on `server.events.on('request')` is only invoked when logging before. There is a GitHub issue where people got confused. Loads of the other internal events are not supported anymore. I went back to use `server.ext` tested it locally and will publish soon.

**Comment by tdeekens** [comment]
Published `@promster/hapi@2.2.5` and tested it locally. Please re-test and feel free to re-open in case it's not solved. Thanks again.

**Comment by 2color** [comment]
Thanks. Works like a breeze
